src/shims/unix/linux/epoll.rs

```diff
@@ -525,3 +508,28 @@ fn ready_list_next(
     }
     return None;
 }
+
+/// This helper function checks whether an epoll notification should be triggered for a specific
+/// epoll_interest and, if necessary, triggers the notification. Unlike check_and_update_readiness,
+/// this function sends a notification to only one epoll instance.
+fn check_fd_ready_event(
```

Comment:
I have a hard time coming up with a good naming to differentiate this function from ``check_and_update_readiness``. They essentially do the same thing, but this function only checks one ``epoll_event_interest``, where ``check_and_update_readiness`` checks all ``epoll_event_interest``. As a result, this function can only return notification to one epoll instance where ``check_and_update_readiness`` can return notification to multiple epoll instances.